src/core/Directus/Application/Http/Middleware/AuthenticationMiddleware.php

```diff
@@ -102,9 +93,15 @@ public function __invoke(Request $request, Response $response, callable $next)
             $hookEmitter->run('auth.fail', [$exception]);
             throw $exception;
         }
-
-
-
+       
+        // TODO: Adding an user should auto set its ID and GROUP
+        // TODO: User data should be casted to its data type
+        // TODO: Make sure that the group is not empty
+        $acl->setUserId($user->getId());
+        $acl->setUserEmail($user->getEmail());
+        $acl->setUserFullName($user->get('first_name') . ' ' . $user->get('last_name'));
+
+        $hookEmitter->run('auth.success', [$user]);
```

Comment:
@itsmerhp 

This code will break when the user has an invalid token. Kindly check that scenario and update the code accordingly. :)

Reply:
@itsmerhp Please refer the issue #1098

Reply:
@bjgajjar @hemratna 
I have committed the change and tested it for the below cases : 

1. Public endpoint
-  Without token
- Valid token of another role user
- Invalid token

2. Non-Public endpoint
- Without token
- Valid token of another role user
- Invalid token